FrameworkSystem/DB/InstalledComponentsDB.py

```diff
@@ -0,0 +1,1017 @@
+                      unInstallationTime = null() ):
+    self.instance = instance
+    self.installationTime = installationTime
+    self.unInstallationTime = unInstallationTime
+
+  def fromDict( self, dictionary ):
+    """
+    Fill the fields of the InstalledComponent object from a dictionary
+    The dictionary may contain the keys: ComponentID, HostID, Instance,
+    InstallationTime, UnInstallationTime
+    """
+
+    self.componentID = dictionary.get( 'ComponentID', self.componentID )
+    self.hostID = dictionary.get( 'HostID', self.hostID )
+    self.instance = dictionary.get( 'Instance', self.instance )
+    self.installationTime = dictionary.get( 'InstallationTime',
+                                            self.installationTime )
+    self.unInstallationTime = dictionary.get( 'UnInstallationTime',
+                                              self.unInstallationTime )
+
+    return S_OK( 'Successfully read from dictionary' )
+
+  def toDict( self, includeComponents = False, includeHosts = False ):
+    """
+    Return the object as a dictionary
+    If includeComponents is True, information about which Components where
+    installed is included
+    If includeHosts is True, information about the Hosts where the
+    installations are is included
+    """
+
+    dictionary = {
+                  'Instance': self.instance,
+                  'InstallationTime': self.installationTime,
+                  'UnInstallationTime': self.unInstallationTime
+                  }
+
+    if includeComponents:
+      dictionary[ 'Component' ] = self.installationComponent.toDict()[ 'Value' ]
+    else:
+      dictionary[ 'ComponentID' ] = self.componentID
+
+    if includeHosts:
+      dictionary[ 'Host' ] = self.installationHost.toDict()[ 'Value' ]
+    else:
+      dictionary[ 'HostID' ] = self.hostID
+
+    return S_OK( dictionary )
+
+class InstalledComponentsDB( object ):
+  """
+  Class used to work with the InstalledComponentsDB database.
+  It creates the tables on initialization and allows inserting, querying,
+  deleting from/to the tables
+  """
+
+  def __init__( self ):
+    self.__initializeConnection( 'Framework/InstalledComponentsDB' )
+    result = self.__initializeDB()
+    if not result[ 'OK' ]:
+      raise Exception( "Can't create tables: %s" % result[ 'Message' ] )
+
+  def __initializeConnection( self, dbPath ):
+
+    result = getDBParameters( dbPath, defaultQueueSize = 10 )
+    if not result[ 'OK' ]:
+      raise Exception \
+                  ( 'Cannot get the Database parameters' % result( 'Message' ) )
+
+    dbParameters = result[ 'Value' ]
+    self.host = dbParameters[ 'host' ]
+    self.user = dbParameters[ 'user' ]
+    self.password = dbParameters[ 'password' ]
+    self.db = dbParameters[ 'db' ]
+
+    self.engine = create_engine( 'mysql://%s:%s@%s/%s' %
+                    ( self.user, self.password, self.host, self.db ),
+                    pool_recycle = 3600, echo_pool = True
+                    )
```

Comment:
I fixed the same thing in RequesDB in https://github.com/fstagni/DIRAC/commit/ebe9a10be1d5debca97b094df97d1fa4e749daaf